File: 02/2606/2606.js
let fs = require("fs");
let [a, b, ...input] = fs.readFileSync("/dev/stdin").toString().split("\n");
let graph = {};

for (let i = 0; i < Number(b); i++) {
  input[i] = input[i].split(" ").map(Number);
  if (graph[input[i][0]] != undefined) graph[input[i][0]].push(input[i][1]);
  else graph[input[i][0]] = [input[i][1]];
  if (graph[input[i][1]] != undefined) graph[input[i][1]].push(input[i][0]);
  else graph[input[i][1]] = [input[i][0]];
}

let willVisit = [1];
let visited = [];
while (willVisit.length != 0) {
  let target = willVisit.shift();
  if (visited.indexOf(target) == -1) {
    visited.push(target);
    willVisit.push(...graph[target]);
  }
}

console.log(visited.length - 1);

File: 02/2606/2606.py
import sys
input = sys.stdin.readline
print = sys.stdout.write

n = int(input())
p = int(input())
data = [list(map(int, input().split(' '))) for _ in range(p)]
tree = {}
for i in range(p):
    if data[i][0] in tree:
        tree[data[i][0]].add(data[i][1])
    else:
        tree[data[i][0]] = set([data[i][1]])
    if data[i][1] in tree:
        tree[data[i][1]].add(data[i][0])
    else:
        tree[data[i][1]] = set([data[i][0]])
if (not 1 in tree):
    print("0")
    exit()
visited = []
willVisit = [1]
while (len(willVisit) != 0):
    now = willVisit.pop()
    if (now in visited):
        continue
    visited.append(now)
    willVisit.extend(list(tree[now]))
print(str(len(visited) - 1))
